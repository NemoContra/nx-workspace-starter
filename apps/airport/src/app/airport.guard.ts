import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
import { Observable } from 'rxjs';

@Injectable()
export class AirportGuard implements CanActivate {
  canActivate(
    context: ExecutionContext
  ): boolean | Promise<boolean> | Observable<boolean> {
    const ctx = context.switchToRpc().getContext();
    const data = context.switchToRpc().getData();
    console.log('GUARD', JSON.parse(ctx.args[1]).cmd, data);

    return data === 'airports';
  }
}
