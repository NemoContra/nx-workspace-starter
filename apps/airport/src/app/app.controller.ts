import { Controller, UseGuards } from '@nestjs/common';
import { MessagePattern } from '@nestjs/microservices';
import { AirportGuard } from './airport.guard';

@Controller()
export class AppController {
  @MessagePattern({ cmd: 'airports' })
  @UseGuards(AirportGuard)
  public getAirports(): string[] {
    return [
      'Flughafen Wien-Schwechat',
      'Flughafen Manfred Rommel Stuttgart',
      'Flughafen Helmut Schmidt Hamburg',
    ];
  }
}
