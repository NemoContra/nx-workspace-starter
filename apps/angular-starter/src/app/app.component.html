<div class="wrapper">
  <div class="sidebar" data-background-color="white" data-active-color="danger">
    <flight-sidebar />
  </div>

  <div class="main-panel">
    <flight-navbar />

    <div class="content">
      <flight-search />
    </div>
  </div>
</div>
