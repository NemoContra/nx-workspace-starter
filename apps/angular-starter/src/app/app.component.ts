import { Component } from '@angular/core';
import { NavbarComponent } from './navbar/navbar.component';
import { SidebarComponent } from './sidebar/sidebar.component';
import { RouterOutlet } from '@angular/router';

@Component({
  selector: 'flight-app',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  standalone: true,
  imports: [NavbarComponent, SidebarComponent, RouterOutlet],
})
export class AppComponent {
  title = 'Hello World!';
}
