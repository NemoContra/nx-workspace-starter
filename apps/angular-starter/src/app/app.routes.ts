import { Routes } from '@angular/router';

export const APP_ROUTES: Routes = [
  {
    path: 'flight-booking',
    loadChildren: () =>
      import('./flight-booking/flight-booking.routes').then(
        (m) => m.FLIGHT_BOOKING_ROUTES
      ),
  },
];
