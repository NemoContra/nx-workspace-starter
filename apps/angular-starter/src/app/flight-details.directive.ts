import {
  Directive,
  HostBinding,
  HostListener,
  Input,
  OnInit,
} from '@angular/core';
import { Flight } from '@flight-app/shared';

@Directive({ selector: '[flightDetails]', standalone: true })
export class FlightDetailsDirective implements OnInit {
  @Input() flightDetails: Flight;
  @HostBinding() title = '';

  @HostListener('click') public onClick(): void {
    alert(`From: ${this.flightDetails.from} to: ${this.flightDetails.to}`);
  }

  ngOnInit(): void {
    if (this.flightDetails) {
      this.title = `From: ${this.flightDetails.from} to: ${this.flightDetails.to}`;
    }
  }
}
