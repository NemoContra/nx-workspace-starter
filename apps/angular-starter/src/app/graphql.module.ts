import { NgModule } from '@angular/core';
import { ApolloModule, APOLLO_OPTIONS } from 'apollo-angular';
import { HttpLink } from 'apollo-angular/http';
import { InMemoryCache } from '@apollo/client/core';

const uri = 'http://localhost:3333/graphql'; // <-- add the URL of the GraphQL server here
export function createApollo(httpLink: HttpLink) {
  return {
    cache: new InMemoryCache(),
    link: httpLink.create({ uri }),
  };
}

@NgModule({
  imports: [ApolloModule],
  providers: [
    {
      provide: APOLLO_OPTIONS,
      useFactory: createApollo,
      deps: [HttpLink],
    },
  ],
})
export class GraphQLModule {}
