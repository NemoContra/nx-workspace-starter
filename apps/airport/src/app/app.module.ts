import { Logger, Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { CacheModule } from '@nestjs/cache-manager';

@Module({
  imports: [
    CacheModule.register({
      ttl: 60,
      max: 10,
    }),
  ],
  controllers: [AppController],
  providers: [
    { provide: Logger, useFactory: () => new Logger('Airport Microservice') },
    AppService,
  ],
})
export class AppModule {}
