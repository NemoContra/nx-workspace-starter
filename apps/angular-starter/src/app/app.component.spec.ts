import { createComponentFactory } from '@ngneat/spectator/jest';
import { AppComponent } from './app.component';
import { provideRouter } from '@angular/router';

describe('AppComponent', () => {
  const createComponent = createComponentFactory({
    component: AppComponent,
    providers: [provideRouter([])],
  });

  it('should render', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
