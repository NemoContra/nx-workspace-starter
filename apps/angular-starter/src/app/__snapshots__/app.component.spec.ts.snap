// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render 1`] = `
<flight-app
  title={[Function String]}
>
  <div
    class="wrapper"
  >
    <div
      class="sidebar"
      data-active-color="danger"
      data-background-color="white"
    >
      <flight-sidebar>
        <div
          class="sidebar-wrapper"
        >
          <div
            class="logo"
          >
            <a
              class="simple-text"
            >
              <div
                class="logo-img"
              >
                <img
                  alt=""
                  src="../../assets/img/angular2-logo.png"
                />
              </div>
               Flight42 
            </a>
          </div>
          <ul
            class="nav"
          >
            <li>
              <a>
                <i
                  class="ti-home"
                />
                <p>
                  <a
                    href="/home"
                    routerlink="home"
                  >
                    Home
                  </a>
                </p>
              </a>
            </li>
            <li>
              <a>
                <i
                  class="ti-arrow-top-right"
                />
                <p>
                  <a
                    href="/flight-search"
                    routerlink="flight-search"
                  >
                    Flights
                  </a>
                </p>
              </a>
            </li>
            <li>
              <a>
                <i
                  class="ti-user"
                />
                <p>
                  <a
                    href="/passenger-search"
                    routerlink="passenger-search"
                  >
                    Passengers
                  </a>
                </p>
              </a>
            </li>
          </ul>
        </div>
      </flight-sidebar>
    </div>
    <div
      class="main-panel"
    >
      <flight-navbar>
        <nav
          class="navbar navbar-default"
        >
          <div
            class="container-fluid"
          >
            <div
              class="navbar-header"
            >
              <button
                class="navbar-toggle"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Toggle navigation
                </span>
                <span
                  class="icon-bar bar1"
                />
                <span
                  class="icon-bar bar2"
                />
                <span
                  class="icon-bar bar3"
                />
              </button>
              <a
                class="navbar-brand"
                href="#"
              >
                Flight42
              </a>
            </div>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="nav navbar-nav navbar-right"
              >
                <li>
                  <a
                    class="dropdown-toggle"
                    data-toggle="dropdown"
                    href="#"
                  >
                    <i
                      class="ti-panel"
                    />
                    <p>
                      Stats
                    </p>
                  </a>
                </li>
                <li
                  class="dropdown"
                >
                  <a
                    class="dropdown-toggle"
                    data-toggle="dropdown"
                    href="#"
                  >
                    <i
                      class="ti-bell"
                    />
                    <p
                      class="notification"
                    >
                      5
                    </p>
                    <p>
                      Notifications
                    </p>
                    <b
                      class="caret"
                    />
                  </a>
                  <ul
                    class="dropdown-menu"
                  >
                    <li>
                      <a
                        href="#"
                      >
                        Notification 1
                      </a>
                    </li>
                    <li>
                      <a
                        href="#"
                      >
                        Notification 2
                      </a>
                    </li>
                    <li>
                      <a
                        href="#"
                      >
                        Notification 3
                      </a>
                    </li>
                    <li>
                      <a
                        href="#"
                      >
                        Notification 4
                      </a>
                    </li>
                    <li>
                      <a
                        href="#"
                      >
                        Another notification
                      </a>
                    </li>
                  </ul>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    <i
                      class="ti-settings"
                    />
                    <p>
                      Settings
                    </p>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </flight-navbar>
      <div
        class="content"
      >
         Hello World! 
        <router-outlet />
      </div>
    </div>
  </div>
</flight-app>
`;
